JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,598 @@
+# React Guidelines and Best Practices
+
+*Guides are not rules and should not be followed blindly. Use your head and think.*
+
+### File Organization and Naming
+As it is a good practice to divide React components as container and presentational components, you should place them in separate folders (this example uses **containers** and **components** for folder naming, but there are other naming conventions, depending on the use case and preference of developers).
+
+```
+...
+├─ components
+├─ containers 
+...
+```
+
+Use **camelCase** for general folder naming and **PascalCase** for React components, their stylesheet files and their containing folders.
+
+```
+...
+├─ components
+⎮   ├─ LoginContent
+⎮   ⎮   └─ LoginContent.js
+⎮   ⎮   └─ LoginContent.scss
+⎮   └─ LoginHeader
+⎮       └─ LoginHeader.js
+⎮       └─ LoginHeader.scss
+├─ containers 
+⎮   └─ Login
+⎮       └─ Login.js
+⎮       └─ Login.scss 
+...
+```
+
+### Component Declaration
+
+For declaring class components:
+- use ES2015 **class syntax**
+- don't use *React.createClass*
```

Comment:
Maybe add a note:
If you absolutely have to, use the [`create-react-class`](https://www.npmjs.com/package/create-react-class) package instead of `React.createClass` (removed in React 16).